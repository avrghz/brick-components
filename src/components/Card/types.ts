export const SHADOW = ['never', 'always', 'hover'] as const

export type Shadow = typeof SHADOW[number]
