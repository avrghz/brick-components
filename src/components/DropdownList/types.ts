export interface Option {
    label: string
    value: string
    disabled?: boolean
}

export type Callback = (index: number) => void
