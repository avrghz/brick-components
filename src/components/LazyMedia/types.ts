export type MediaType = 'img' | 'picture'
