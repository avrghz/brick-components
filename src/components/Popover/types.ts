export const TRIGGER_ON = ['click', 'focus', 'hover', 'manual'] as const

export type TriggerOn = typeof TRIGGER_ON[number]
