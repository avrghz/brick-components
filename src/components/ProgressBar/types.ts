export const BAR_TYPE = ['line', 'circle', 'dashboard'] as const

export type BarType = typeof BAR_TYPE[number]

export interface ProgressProps {
    progress: number
    showPercentageText?: boolean
    textInside?: boolean
}

export interface ProgressTextProps extends Pick<ProgressProps, 'progress'> {
    show?: boolean
    isInnerText?: boolean
}

export interface ProgressCircleProps extends ProgressProps {
    strokeDash: [number, number]
    strokeDashOffset: number
    flip?: boolean
}
