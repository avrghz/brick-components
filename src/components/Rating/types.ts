export interface StarProps {
    hover?: boolean
    activeColor?: string
}

export interface Colors {
    [key: number]: string
}
