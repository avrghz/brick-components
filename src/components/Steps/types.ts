import { JSXBase } from '@stencil/core/internal'

export const STEP_STATE = ['process', 'wait', 'success', 'error', 'finish'] as const

export interface StepComponent {
    step: number
    title: string
    description?: string
    state?: typeof STEP_STATE[number]
}

export interface StepProps extends StepComponent {
    style?: JSXBase.HTMLAttributes<HTMLDivElement>['style']
    isLast?: boolean
    isCentered?: boolean
}
