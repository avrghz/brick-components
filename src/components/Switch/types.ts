export interface AVChange {
    isOn: boolean
}
