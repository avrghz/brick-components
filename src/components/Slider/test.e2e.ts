import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'
import { VARIANTS } from '../../shared/types'

describe('<bk-slider/>', () => {
    let page: E2EPage
    let component: E2EElement

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-slider/>`)
        component = await page.find('bk-slider')
    })

    const getDragButton = async () => await page.find('bk-slider >>> .bk-slider__button')
    const getTooltip = async () => await page.find('bk-slider >>> bk-popover')
    const getSlider = async () => await page.find('bk-slider >>> .bk-slider')

    it('should render', async () => {
        expect(component).not.toBeNull()
    })

    describe('Prop: disabled', () => {
        it('should render as disabled, when set to true', async () => {
            component.setProperty('disabled', true)
            await page.waitForChanges()
            expect(await getSlider()).toHaveClass(`bk-slider--disabled`)
        })

        it('should render as not disabled, when set to false', async () => {
            component.setProperty('disabled', false)
            await page.waitForChanges()
            expect(await getSlider()).not.toHaveClass(`bk-slider--disabled`)
        })
    })

    describe('Prop: variant', () => {
        VARIANTS.forEach((variant) => {
            it(`it should render as ${variant}`, async () => {
                component.setProperty('variant', variant)
                await page.waitForChanges()
                expect(await getSlider()).toHaveClass(`bk-slider--${variant}`)
            })
        })
    })

    describe('Prop: showTooltip', () => {
        it('should not show tooltip, when set to false', async () => {
            component.setProperty('showTooltip', false)
            await page.waitForChanges()
            const dragButton = await getDragButton()
            dragButton.focus()
            await page.waitForChanges()
            expect(await getTooltip()).toBeNull()
        })

        it('should not show tooltip, when set to true', async () => {
            component.setProperty('showTooltip', true)
            await page.waitForChanges()
            const dragButton = await getDragButton()
            dragButton.focus()
            await page.waitForChanges()
            expect(await getTooltip()).not.toBeNull()
        })
    })

    describe('Prop: value', () => {
        it('should render with given value', async () => {
            component.setProperty('value', 50)
            await page.waitForChanges()
            await page.waitForChanges()
            expect(component).toMatchSnapshot()
        })

        it('should be able to click on bar to set the value', async () => {
            await (await getSlider()).click()
            await page.waitForChanges()
        })
    })

    describe('Prop: step', () => {
        it('should move by given `step`', async () => {
            const bkChange = await component.spyOnEvent('bkChange')

            component.setProperty('step', 10)
            await page.waitForChanges()
            const dragButton = await getDragButton()
            const tooltip = await getTooltip()

            await dragButton.press('ArrowUp')
            await page.waitForChanges()
            expect(tooltip.innerText).toBe('10')

            await dragButton.press('ArrowRight')
            await page.waitForChanges()
            expect(tooltip.innerText).toBe('20')

            await dragButton.press('ArrowDown')
            await page.waitForChanges()
            expect(tooltip.innerText).toBe('10')

            await dragButton.press('ArrowLeft')
            await page.waitForChanges()
            expect(tooltip.innerText).toBe('0')

            expect(bkChange).toHaveNthReceivedEventDetail(0, 10)
            expect(bkChange).toHaveNthReceivedEventDetail(1, 20)
            expect(bkChange).toHaveNthReceivedEventDetail(2, 10)
            expect(bkChange).toHaveNthReceivedEventDetail(3, 0)
        })
    })
})
