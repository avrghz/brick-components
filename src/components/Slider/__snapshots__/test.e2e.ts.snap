// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-slider/> Prop: value should render with given value 1`] = `
<bk-slider aria-label="slider between 0 and 100" aria-value="50" aria-valuemax="100" aria-valuemin="0" class="hydrated" role="slider" value="50">
  <mock:shadow-root>
    <div class="bk-slider bk-slider--default">
      <div class="bk-slider__bar" style="width: 50%;"></div>
      <div class="bk-slider__button" tabindex="0" style="left: 50%;">
        <bk-popover aria-label="popover" class="hydrated">
          <div class="bk-slider__button-inner" slot="control"></div>
          <span class="bk-slider__tooltip-content" slot="content">
            50
          </span>
        </bk-popover>
      </div>
    </div>
  </mock:shadow-root>
</bk-slider>
`;
